### PR TITLE
Added Surface Extensibility support for Xaml Behaviors

### PR DESCRIPTION
**Pull request by hemantbajpai**
### Description of Change ###

This PR adds DesignTools dll to provide support for surface extensibility. Now the DesignTools dll will get build and added to nuget package using nuspec file. DesignTools dll adds various attributes to xaml behaviors.  

### Bugs Fixed ###

N/A

### API Changes ###

New dll is added to support surface extensibility, nothing is modified in existing APIs.

Added:
 - DesignTools dll
-  Nuspec file to pack both dlls

Changed:
 - Microsoft.Xaml.Behaviors.csproj

### Behavioral Changes ###

Now in wpf dot net core user can see various attributes functionality added to behavior.

### CODE REVIEW

File: src/BehaviorsSdk.sln [hunk 1]
```diff
@@ -1,9 +1,9 @@
 ﻿
 Microsoft Visual Studio Solution File, Format Version 12.00
-# Visual Studio 15
-VisualStudioVersion = 15.0.26222.1
+# Visual Studio Version 16
+VisualStudioVersion = 16.0.28829.227
 MinimumVisualStudioVersion = 10.0.40219.1
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Microsoft.Xaml.Behaviors", "Microsoft.Xaml.Behaviors\Microsoft.Xaml.Behaviors.csproj", "{8620C598-B704-429E-BCA5-7E5DDE506923}"
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Microsoft.Xaml.Behaviors", "Microsoft.Xaml.Behaviors\Microsoft.Xaml.Behaviors.csproj", "{8620C598-B704-429E-BCA5-7E5DDE506923}"
```
Comment (branh) [inline src/BehaviorsSdk.sln]: Why did this project GUID change? What does this mean?

File: src/Microsoft.Xaml.Behaviors.DesignTools/packages.config [hunk 2]
```diff
@@ -0,0 +1,4 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<packages>
+  <package id="NuGet.Build.Tasks.Pack" version="5.0.0" targetFramework="net461" developmentDependency="true" />
```
Comment (branh) [inline src/Microsoft.Xaml.Behaviors.DesignTools/packages.config]: You added the NuGet.Build.Tasks.Pack NuGet package to Microsoft.Xaml.Behaviors.DesignTools, but the pack target itself is in Microsoft.Xaml.Behaviors so this NuGet package should be too.

File: src/Microsoft.Xaml.Behaviors.DesignTools/KnownUnreferencedTypeAttribute.cs [hunk 3]
```diff
@@ -0,0 +1,19 @@
+﻿using System;
```
Comment (branh) [inline src/Microsoft.Xaml.Behaviors.DesignTools/KnownUnreferencedTypeAttribute.cs]: Should have the MIT license header.

File: src/Microsoft.Xaml.Behaviors.DesignTools/Microsoft.Xaml.Behaviors.DesignTools.csproj [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    <Optimize>false</Optimize>
+    <OutputPath>..\Microsoft.Xaml.Behaviors\bin\Debug\net45\</OutputPath>
+    <DefineConstants>DEBUG;TRACE</DefineConstants>
+    <ErrorReport>prompt</ErrorReport>
+    <WarningLevel>4</WarningLevel>
+  </PropertyGroup>
+  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Release|AnyCPU' ">
+    <DebugType>pdbonly</DebugType>
+    <Optimize>true</Optimize>
+    <OutputPath>bin\Release\</OutputPath>
```
Comment (branh) [inline src/Microsoft.Xaml.Behaviors.DesignTools/Microsoft.Xaml.Behaviors.DesignTools.csproj]: Why are the output paths for debug and release so different?

File: Microsoft.Xaml.Behaviors.nuspec [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+      <frameworkAssembly assemblyName="PresentationFramework" targetFramework=".NETFramework4.5" />
+      <frameworkAssembly assemblyName="System.Core" targetFramework=".NETFramework4.5" />
+      <frameworkAssembly assemblyName="System" targetFramework=".NETFramework4.5" />
+      <frameworkAssembly assemblyName="System.Xaml" targetFramework=".NETFramework4.5" />
+      <frameworkAssembly assemblyName="System.XML" targetFramework=".NETFramework4.5" />
+      <frameworkAssembly assemblyName="WindowsBase" targetFramework=".NETFramework4.5" />
+    </frameworkAssemblies>
+  </metadata>
+    <files>
+    <file src="src\Microsoft.Xaml.Behaviors\bin\Debug\net45\Microsoft.Xaml.Behaviors.pdb" target="lib\net45\Microsoft.Xaml.Behaviors.pdb" />
```
Comment (branh) [inline Microsoft.Xaml.Behaviors.nuspec]: Shouldn't we be using the release versions in the NuGet package?

File: src/Microsoft.Xaml.Behaviors.DesignTools/Properties/AssemblyInfo.cs [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+[assembly: AssemblyTrademark("")]
+[assembly: AssemblyCulture("")]
+
+// Setting ComVisible to false makes the types in this assembly not visible
+// to COM components.  If you need to access a type in this assembly from
+// COM, set the ComVisible attribute to true on that type.
+[assembly: ComVisible(false)]
+
+// The following GUID is for the ID of the typelib if this project is exposed to COM
+[assembly: Guid("6c3f3432-d090-450b-822b-95b7c3cef663")]
```
Comment (branh) [inline src/Microsoft.Xaml.Behaviors.DesignTools/Properties/AssemblyInfo.cs]: I realize this is probably all auto-generated, but simply based off the comments, if ComVisible is set to false what is the point of the Guid?

File: src/Microsoft.Xaml.Behaviors.DesignTools/UnreferencedTypeId.cs [hunk 7]
```diff
@@ -0,0 +1,36 @@
+﻿using System;
```
Comment (branh) [inline src/Microsoft.Xaml.Behaviors.DesignTools/UnreferencedTypeId.cs]: MIT license header?

File: src/Microsoft.Xaml.Behaviors.DesignTools/MetadataTableProvider.cs [hunk 8]
```diff
@@ -0,0 +1,348 @@
+﻿using Microsoft.VisualStudio.DesignTools.Extensibility.Metadata;
```
Comment (branh) [inline src/Microsoft.Xaml.Behaviors.DesignTools/MetadataTableProvider.cs]: MIT license header?

**Comment by lutzroeder** [comment]
`src/Microsoft.Xaml.Behaviors.DesignTools/Preprocess.xml` shouldn't be part of this change?

**Comment by onovotny** [comment]
If you need to change the pool to 2019, you can do that by updating the `azure-pipelines.yml` in the PR. Change this to `windows-2019`: https://github.com/microsoft/XamlBehaviorsWpf/blob/master/azure-pipelines.yml#L10

**Comment by brianlagunas** [comment]
Thanks for the design tools 👯‍♂

**Pull request merged**